HW2/P5/driver.py

```diff
@@ -80,3 +96,21 @@ def randcolor():
             # SUBPROBLEM 3: sort objects by location.  Be sure to update the
             # grid if objects' indices change!  Also be sure to sort the
             # velocities with their object positions!
+            
+            # Get Positions and sort the points according to morton ordering
+            grid_positions = (positions / grid_spacing).astype(int) # Maintain grid spacing
+            zOrderCurve = map(lambda x, y: cmp_zorder(x,y), grid_positions, np.array(list(grid_positions[1:]) + list(grid_positions[:1])))
```

Comment:
I looked at the same cmp_zorder from the wikipedia page when I was working on my own code and my understanding was that the cmp_zorder function only provided relatives values, i.e. it was a shortcut to figure out which of the two given elements passed in has a higher z-order. Your code offsets the list and compares each element to the one next to it, then sorts by this relative value. This could provide approximately the right answer but I am not sure if it completely correct because the value each element is mapped to depends on the element adjacent to it.